src/shared/utils/adler32.js

```diff
@@ -13,17 +13,35 @@
 
 var MOD = 65521;
 
-// This is a clean-room implementation of adler32 designed for detecting
-// if markup is not what we expect it to be. It does not need to be
-// cryptographically strong, only reasonably good at detecting if markup
-// generated on the server is different than that on the client.
+// adler32 is not cryptographically strong, and is only used to sanity check that
+// markup generated on the server matches the markup generated on the client.
+// This implementation (a modified version of the SheetJS version) has been optimized
+// for our use case, at the expense of conforming to the adler32 specification
+// for non-ascii inputs.
 function adler32(data) {
   var a = 1;
   var b = 0;
-  for (var i = 0; i < data.length; i++) {
-    a = (a + data.charCodeAt(i)) % MOD;
-    b = (b + a) % MOD;
+  var i;
+  var l = data.length;
+  var m = l & 0xFFFFFFFC;
+  for (i = 0; i < m; ) {
```

Comment:
Just a nit,  but why not just `while (i < m) {`  and initialize `i` earlier?
`for` loops with expressions missing always just looks odd IMHO.